### PR TITLE
feat: support multiple yaml schema version

### PR DESCRIPTION
**Pull request by 1yefuwang1**
https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/24088209
This PR adds support for multiple yaml schema versions, and bump latest yaml schema version to 1.1.0

For invalid versions, the following error message will be shown
image

### CODE REVIEW

**Review by jayzhang — APPROVED** [review]

File: packages/fx-core/resource/package.nls.json [hunk 1]
```diff
@@ -615,6 +615,7 @@
   "driver.botFramework.progressBar.createOrUpdateBot": "Creating or updating bot registration.",
   "error.yaml.InvalidYamlSchemaError": "Unable to parse yaml file: %s. Please review the yaml file.",
   "error.yaml.InvalidYamlSchemaErrorWithReason": "Unable to parse yaml file: %s. Reason: %s Please review the yaml file.",
+  "error.yaml.VersionNotSupported": "version %s is not supported. Supported versions: %s.",
```
Comment (sffamily) [inline packages/fx-core/resource/package.nls.json:618]: does the first letter need to be capitalized?    version -> Version

Comment (1yefuwang1) [reply packages/fx-core/resource/package.nls.json:618]: It's part of another error message. 
image

**Review by sffamily — APPROVED** [review]

**Review by LongOddCode — APPROVED** [review]

**Review by MSFT-yiz — APPROVED** [review]

**Review by wenytang-ms — APPROVED** [review]

File: packages/fx-core/src/component/driver/interface/stepDriver.ts [hunk 2]
```diff
@@ -27,10 +27,12 @@ export interface StepDriver {
...
    * @param ctx logger, telemetry, progress bar, etc.
    * @param outputEnvVarNames the environment variable names for each output
+   * @param schemaVersion schema version of the executed yaml file
    */
   execute?(
     args: unknown,
     ctx: DriverContext,
-    outputEnvVarNames?: Map<string, string>
+    outputEnvVarNames?: Map<string, string>,
+    schemaVersion?: string
```
Comment (kuojianlu) [inline packages/fx-core/src/component/driver/interface/stepDriver.ts:36]: Action should also support different versions, right?

Comment (1yefuwang1) [reply packages/fx-core/src/component/driver/interface/stepDriver.ts:36]: Yeah, now actions know which schema version they are operating on.

**Pull request merged**